src/services/product.service.ts

```diff
@@ -46,4 +54,30 @@ const getFiltered = async (query: string) => {
   return products;
 };
 
-export default { getAll, getOne, getFiltered };
+function getRandom(products: IProduct[], limit: number) {
+  const randomProducts = [];
+  const indexes: number[] = [];
+  let i = 0;
+
+  while (i < limit) {
+    const randomIndex = Math.floor(Math.random() * products.length);
+
+    if (indexes.includes(randomIndex)) {
+      continue;
+    } else {
+      indexes.push(randomIndex);
+      i++;
+      randomProducts.push(products[randomIndex]);
+    }
+  }
+
+  return randomProducts;
+}
+
+export default {
+  allProducts,
```

Comment:
You won't need to export allProducts, it will be used only inside service

Reply:
let it be for now